modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/collateral/CollateralService.scala

```diff
@@ -0,0 +1,64 @@
+package org.tessellation.sdk.infrastructure
+
+import cats.effect.std.Queue
+import cats.effect.{Async, Ref, Spawn}
+import cats.syntax.contravariantSemigroupal._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.Balance
+import org.tessellation.sdk.domain.collateral.CollateralService
+
+import eu.timepit.refined.auto._
+import eu.timepit.refined.types.numeric
+import fs2.Stream
+
+object CollateralService {
+
+  def make[F[_]: Async]: F[CollateralService[F]] = {
+    def mkProperCollateral = Ref.of[F, Set[Address]](Set.empty)
+    def mkOffloadQueue = Queue.unbounded[F, (Long, Map[Address, Balance])]
+    (mkProperCollateral, mkOffloadQueue).mapN(make(_, _)).flatten
+  }
+
+  def make[F[_]: Async](
+    properBalancesCache: Ref[F, Set[Address]],
+    updateBalanceQueue: Queue[F, (Long, Map[Address, Balance])]
+  ): F[CollateralService[F]] = {
+
+    def processUpdates =
+      Stream
+        .fromQueueUnterminated(updateBalanceQueue)
+        .filterWithPrevious {
+          case ((previousNo, _), (currentNo, _)) => previousNo < currentNo
+        }
+        .map {
+          case (_, balances) => balances
+        }
+        .evalMap(incomingBalances => {
+          val (improperBalances, properBalances) =
+            incomingBalances.partition { case (_, balance) => balance.value < numeric.NonNegLong(250000000L) }
```

Comment:
please don't hardcode such values, it should be parametrized and taken from configuration.
Or - written in genesis block and consecutive snapshots like "network metadata". What you guys think about that? @tbekas @kpudlik @TheMMaciek

Having "network metadata" we could enforce using the same values across the network and having it in snapshot would allow to change it in the go. I'm concerned about repeating the same values over and over again 🤔 any better idea?

Reply:
Yeah, I was initially thinking about just lifting that value in the file so it is not in between the lines but keeping it in genesis is also fine.

Reply:
On the other hand repeating it in every snapshot is redundant so maybe hardcoding in the code is better.

Reply:
Threshold value extracted to `CollateralConfig` as `minStakingAmount`